flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1407,14 +1412,23 @@ private void failPendingCheckpoint(
 		failPendingCheckpoint(pendingCheckpoint, reason, null);
 	}
 
+	private void failPendingCheckpoint(
+		final PendingCheckpoint pendingCheckpoint,
+		final CheckpointFailureReason reason,
+		final ExecutionAttemptID executionAttemptID) {
+
+		failPendingCheckpoint(pendingCheckpoint, reason, null, executionAttemptID);
+	}
+
 	private void failPendingCheckpoint(
 			final PendingCheckpoint pendingCheckpoint,
 			final CheckpointFailureReason reason,
-			final Throwable cause) {
+			final Throwable cause,
+			final ExecutionAttemptID executionAttemptID) {
```

Comment:
you are missing `@Nullable` annotation.

Also per the recent `[DISCUSS][CODE STYLE] Usage of Java Optional` dev mailing list discussion, I would actually prefer to make this `cause ` and `executionAttemptID` parameters `Optional` (it's a private method) as this would avoid this missing `@Nullable` annotation errors.

Reply:
It's suggested to not using `Optional` as parameters, please refer to my [reply](https://s.apache.org/nbl83) here. And yes will add `@Nullable` annotation here.